apps/src/applab/assetManagement/assetListStore.js

```diff
@@ -0,0 +1,26 @@
+var assets = [];
+
+module.exports = {
+  reset: function (list) {
+    assets = list;
+    return assets;
+  },
+
+  add: function (asset) {
+    assets.push(asset);
+    return assets;
+  },
+
+  remove: function (filename) {
+    assets = assets.filter(function (asset) {
+      return asset.filename !== filename;
+    });
+    return assets;
+  },
+
+  list: function (typeFilter) {
+    return typeFilter ? assets.filter(function (asset) {
+      return asset.category === typeFilter;
+    }) : assets;
+  }
+};
```

Comment:
Cool. If you wanted more immutability, you could clone the list before returning it each time (as well as before assigning it in reset). This would mean you don't have to worry about callers modifying your store state. Might be overkill for this context.